packaging/rpm/SPECS/apache-ignite.spec

```diff
@@ -25,14 +34,113 @@ BuildArch:        noarch
 
+This package is a virtual aggregation package for all packages for current
+Apache Ignite release - install it to install all other packages
+
+
+%package -n %{nameCore}
+Summary:          Apache Ignite execution core 
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Requires:         java-1.8.0, pkgconfig
+Requires(pre):    shadow-utils
+Provides:         %{name}-core
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
+%description -n %{nameCore}
+Apache Ignite's core files and libs, necessary for node operation
+
 
+%package -n %{nameLibs}
+Summary:          Apache Ignite optional libs
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Requires:         %{name}-core = %{version}-%{release}
+Provides:         %{name}-libs
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
+%description -n %{nameLibs}
+Apache Ignite's optional libs and integrations
+
+
+%package -n %{nameBenchmarks}
+Summary:          Apache Ignite benchmarks
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Requires:         %{name}-core = %{version}-%{release}
+Provides:         %{name}-benchmarks
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
+%description -n %{nameBenchmarks}
+Prepared maven project for Apache Ignite's benchmarks conduct
+
+
+%package -n %{nameExamples}
+Summary:          Apache Ignite examples
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Provides:         %{name}-examples
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
+%description -n %{nameExamples}
+Apache Ignite's integration and usage examples
+
+
+%package -n %{nameDocs}
+Summary:          Apache Ignite documentation
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Provides:         %{name}-docs
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
+%description -n %{nameDocs}
+Apache Ignite's Javadoc and Scaladoc documentation
+
+
+%package -n %{nameCpp}
+Summary:          Apache Ignite C++ platform
+Version:          %{version}
+Group:            Development/System
+License:          ASL 2.0
+URL:              https://ignite.apache.org
+Obsoletes:        apache-ignite < 2.5.0
+Provides:         %{name}-cpp
+AutoReq:          no
+AutoProv:         no
+BuildArch:        noarch
```

Comment:
That's a problem here. What the platform exactly they are for? Is there native code?

Reply:
There are sources and headers which require preliminary compilation